File: playground/notebook/python/convex_hull.py
# %% [markdown]
# 凸包(Convex Hull)の計算
#
# https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.ConvexHull.html

# %%
from scipy.spatial import ConvexHull
import juliapkg
import juliacall
import matplotlib.pyplot as plt

# %%
jl = juliacall.Main
jl.seval("using Random")

# %%
points = jl.rand(jl.Xoshiro(0), 30, 2).to_numpy()
hull = ConvexHull(points)

# %%
plt.plot(points[:, 0], points[:, 1], "o")
for simplex in hull.simplices:
    plt.plot(points[simplex, 0], points[simplex, 1], "k-")

File: assets/convexhull.jl
using Random
using PythonCall
using OffsetArrays: Origin
using PythonPlot: pyplot as plt

ConvexHull = PythonCall.pyimport("scipy.spatial").ConvexHull
points = Origin(0, 0)(rand(Xoshiro(0), 30, 2))
hull = ConvexHull(points)
fig, ax = plt.subplots()
ax.plot(points[:, 0], points[:, 1], "ro")
for pysimplex in hull.simplices
    # Py -> PyArray 
    simplex = PyArray(pysimplex)
    ax.plot(points[simplex, 0], points[simplex, 1], "k-")
end
fig.savefig("plot1.png")
nothing
